### ISSUE TITLE
Use MySQL triggers instead of hashing `data_values` table

### ISSUE THREAD

**Issue by Marigold** [owid/etl#151]
## Problem

In backporting we're currently calculating hash of all `data_values` for a dataset to know whether it needs to be updated. This is rather slow (12mins for all datasets) and it would be better to do this check when writing to `data_values` table rather than on-demand.

## Solution

Use triggers MySQL triggers to update `dataUpdatedAt` timestamp in `variables` table.

An [example of a trigger](https://stackoverflow.com/questions/4753878/how-to-program-a-mysql-trigger-to-insert-row-into-another-table)
```
delimiter #
create trigger variable_trigger after insert on data_values
for each row
begin
    insert into variables (variableId, updatedAt) values (variableId, NOW())
    on duplicate key update
        dataUpdatedAt = VALUES(updatedAt)
end#
```

## Scope

* [ ] make sure triggers don't kill performance when inserting large volume to `data_values`
* [ ] write a TS migration in grapher to add that column
* [ ] add trigger to staging / live DB
* [ ] add new column `dataUpdatedAt` to variable metadata so that it gets picked up by hash calculated from `config`

**Comment by Marigold** [comment]
We already have `dataUpdatedAt` field in `variables` with timestamp of last manual update. We could perhaps reuse that instead of creating triggers.

**Comment by Marigold** [comment]
I looked into `dataUpdatedAt` field and it seems that all services that are modifying data in grapher are updating the field correctly. Services I've checked:

- manual uploads through admin
- automatic uploads in importers
- (ETL inserts to grapher)

I also went through some ad-hoc datasets (covid etc.) and all looked ok.

This means it'll be enough to compare checksum of dataset & variables metadata of which `dataUpdatedAt` is part of, so any updates should trigger backport.
